internal/playlist_creator/DirectoryReader.go

```diff
@@ -0,0 +1,64 @@
+package playlist_creator
+
+import (
+	"awesomeProject/internal/playlist_creator/config"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+)
+
+func ReadFiles(config *config.Config) *FileData {
+	var fileData = FileData{}
+
+	readFilesRecursively(config, config.InputPath, &fileData)
+
+	sort.Slice(fileData.FilesList, func(i, j int) bool {
+		return fileData.FilesList[i].FileName < fileData.FilesList[j].FileName
+	})
+
+	return &fileData
+}
+
+func readFilesRecursively(config *config.Config, directoryNameAndPath string, fileData *FileData) {
+	files, err := os.ReadDir(directoryNameAndPath)
+	if err != nil {
+		log.Fatal(err)
+	}
```

Comment:
since Go does not provide stack trace information in the `err`, it's hard to know what line of code had the error.
usually i would return a [wrapped error](https://go.dev/blog/go1.13-errors#wrapping-errors-with-w):
```suggestion
func readFilesRecursively(config *config.Config, directoryNameAndPath string, fileData *FileData) error {
	files, err := os.ReadDir(directoryNameAndPath)
	if err != nil {
		err = fmt.Errorf("cannot read directory %s: %w", directoryNameAndPath, err)
		return err
	}
```
and explicitly handling errors at each call is usually the Go's way of dealing with errors. (i know, it can be painfully verbose.)

or some useful information to find the location of the problem.
```suggestion
func readFilesRecursively(config *config.Config, directoryNameAndPath string, fileData *FileData) error {
	files, err := os.ReadDir(directoryNameAndPath)
	if err != nil {
		log.Fatalf("cannot read directory %s: %s", directoryNameAndPath, err.Error())
	}
```